File: google/test/src/utils/TestFileHandler.java
package test.src.utils;

import static org.junit.jupiter.api.Assertions.*;

import java.util.*;
import java.util.HashSet;

import org.junit.jupiter.api.Test;

import main.src.core.structures.Document;
import main.src.core.structures.Token;
import main.src.utils.FileHandler;


public class TestFileHandler {

    final public static String RESOURCES_ABSOLUTE_PATH = "D:/projects/internship/codestar/Team8-Codes/google/test/resources/input";

    @Test
    public void testGetDocumentTokens() {
        Document doc = new Document("doc1.txt", RESOURCES_ABSOLUTE_PATH + '/' + "doc1.txt");
        HashSet<Token> expected = new HashSet<>();
        expected.add(new Token("hello"));
        expected.add(new Token("world"));
        HashSet<Token> actual = FileHandler.getDocumentTokens(doc);

        assertEquals(expected, actual);
    }

    @Test
    public void testGetFolderDocuments(){
        ArrayList<Document> expected = new ArrayList<>();
        expected.add(new Document("doc1.txt", "simple/Path"));
        expected.add(new Document("doc2.txt", "simple/Path"));
        expected.add(new Document("doc3.txt", "simple/Path"));
        ArrayList<Document> actual = FileHandler.getFolderDocuments(RESOURCES_ABSOLUTE_PATH);

        assertEquals(expected, actual);
    }

    @Test
    public void testGetFileContent(){
        String expected = "hello world";
        String actual = FileHandler.getFileContent(RESOURCES_ABSOLUTE_PATH + "/doc1.txt");
        assertEquals(expected, actual);
    }
}

File: GoogleSharp/GoogleSharp.Test/Tests/src/utils/TestFileHandler.cs
using System;
using Xunit;
using Utils;
using Structures;
using System.Collections.Generic;

namespace GoogleSharp.Test
{
    public class TestFileHandler
    {
        public const string RESOURCE_PATH = "../../../Tests/resources/input";

        [Fact]
        public void TestGetDocumentsFromFolder()
        {
            var expected = new List<Document>();
            expected.Add(new Document("doc1.txt", "dummyPath/doc1.txt"));
            expected.Add(new Document("doc2.txt", "dummyPath/doc2.txt"));
            expected.Add(new Document("doc3.txt", "dummyPath/doc3.txt"));

            var actual = FileHandler.GetDocumentsFromFolder(RESOURCE_PATH);

            Assert.Equal(expected[0].Id, actual[0].Id);
            Assert.Equal(expected[1].Id, actual[1].Id);
            Assert.Equal(expected[2].Id, actual[2].Id);
        }

        [Fact]
        public void TestGetFileTokens()
        {
            var actual = FileHandler.GetFileTokens(RESOURCE_PATH + "/doc1.txt");
            var expected = new HashSet<Token>();
            expected.Add(new Token("first"));
            expected.Add(new Token("second"));
            expected.Add(new Token("thidr"));
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void TestGetFileContent()
        {
            var actual = FileHandler.GetFileContent(RESOURCE_PATH + "/doc2.txt");
            var expected = "hello world";
            Assert.Equal(expected, actual);
        }
    }
}
